src/caffe/layers/data_layer.cpp

```diff
@@ -112,11 +118,12 @@ void DataLayer<Dtype>::load_batch(Batch<Dtype>* batch) {
   for (int item_id = 0; item_id < batch_size; ++item_id) {
       timer.Start();
       // Get a datum
-      Datum* datum = (reader_.full().pop("Waiting for data"));
+      string* str = (reader_.full().pop("Waiting for data"));
```

Comment:
question: how do you handle the case where you can't load a full batch (when the batch size does not divide the number of items in the database)?